### PR TITLE
reset undo buffer and clipboard on new or close plot

### LINKED ISSUE #4512: While resetting, D-Lite should forget all undo/redo action
When user selects New action or resetting, all undo and redo actions should be removed.

But currently when user resets, undo/redo actions are still enabled,(remembers the previous actions before resetting).

### Steps to reproduce :
1. Load BULK_BLUE_TRACKS.rep
2. Remove all tracks in this file 
4. Select New to reset D-Lite
4. See that undo action is still enabled. Here it should forget every action after resetting.
5. Select undo action, it retrieves the deleted tracks.

### PR DESCRIPTION
**Pull request by ayeshare**
Fixes #4512

### CODE REVIEW

File: org.mwc.cmap.legacy/src/MWC/GUI/Undo/UndoBuffer.java [hunk 1]
```diff
@@ -248,4 +248,10 @@ public final String undoLabel()
 
     return res;
   }
+  public void resetBuffer() {
+    theActions.clear();
+    presentAction=-1;
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/src/MWC/GUI/Undo/UndoBuffer.java]: @ayeshare - I'm nervous that "-1" is a magic number [1]

I've just looked at the code.  `presentAction` doesn't get initialised. I _think_ our code logic means that would not be an issue, but I'd be happier if it did get assigned something.

So, to solve both of the above I suspect we should introduce a private static member field title `UNSET`, which takes a value of -1.

We can then initialise `presentAction` using it, and we can also set the value using it `resetBuffer`



[1]. https://en.wikipedia.org/wiki/Magic_number_(programming)#Unnamed_numerical_constants

Comment (ayeshare) [reply org.mwc.cmap.legacy/src/MWC/GUI/Undo/UndoBuffer.java]: this is done now.

File: org.mwc.cmap.legacy/src/MWC/GUI/Undo/UndoBuffer.java [hunk 2]
```diff
@@ -86,6 +86,7 @@
 {
   static protected final int undo = 1;
   static protected final int redo = 2;
+  private static final int UNSET = -1;
   private final Vector<Action> theActions;
   private int presentAction;
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/src/MWC/GUI/Undo/UndoBuffer.java]: @ayeshare - you should also initialise `presentAction` to `UNSET`

Comment (ayeshare) [reply org.mwc.cmap.legacy/src/MWC/GUI/Undo/UndoBuffer.java]: done

**Comment by IanMayo** [comment]
@helenayele - could you re-test this, please?

**Comment by helenayele** [comment]
Sure Ian ,Testing it..

**Comment by IanMayo** [comment]
@helenayele - I'm happy to merge this, once you've tested it.

**Pull request merged**